src/UI/Art/Animation/SimpleMoveAnimation.java

```diff
@@ -0,0 +1,93 @@
+package UI.Art.Animation;
+
+import java.awt.Graphics;
+import Engine.Path;
+import Engine.XYCoord;
+import Units.Unit;
+
+public class SimpleMoveAnimation implements GameAnimation
+{
+  long startTime = 0;
+
+  private final long maxTime = 250;
+  private long endTime;
+
+  private final Unit actor;
+  private final Path path;
+  private final double tilesPerMs;
+
+  public SimpleMoveAnimation(Unit actor, Path path)
+  {
+    this.actor = actor;
+    this.path = path;
+    startTime = System.currentTimeMillis();
+    tilesPerMs = actor.model.movePower / (double) maxTime;
+    endTime = (long) ((path.getPathLength() - 1) / tilesPerMs);
```

Comment:
This makes it finish immediately for wait-in-place actions. Much better.

Reply:
Pretty happy about how animations have a reasonable (and hard-limit) max time, and faster units animate faster.